presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/BingTileUtils.java

```diff
@@ -168,4 +186,163 @@ private static int axisToCoordinates(double axis, long mapSize)
         int tileAxis = (int) clip(axis * mapSize, 0, mapSize - 1);
         return tileAxis / TILE_PIXELS;
     }
+
+    private static List<BingTile> findRawTileCovering(OGCGeometry ogcGeometry, int maxZoom)
+    {
+        Envelope envelope = getEnvelope(ogcGeometry);
+        Optional<List<BingTile>> trivialResult = handleTrivialCases(envelope, maxZoom);
+        if (trivialResult.isPresent()) {
+            return trivialResult.get();
+        }
+
+        accelerateGeometry(
+                ogcGeometry, OperatorIntersects.local(), Geometry.GeometryAccelerationDegree.enumMedium);
+
+        Deque<TilingEntry> stack = new ArrayDeque<>();
+        Consumer<BingTile> addIntersecting = tile -> {
+            TilingEntry tilingEntry = new TilingEntry(tile);
+            if (satisfiesTileEdgeCondition(envelope, tilingEntry)
+                    && ogcGeometry.intersects(tilingEntry.polygon)) {
+                stack.push(tilingEntry);
+            }
+        };
+
+        // Populate with initial tiles.  Since there aren't many low zoom tiles,
+        // and throwing away totally disjoint ones is cheap (envelope check),
+        // we might as well start comprehensively.
+        ImmutableList.of(
+                BingTile.fromCoordinates(0, 0, 1),
+                BingTile.fromCoordinates(0, 1, 1),
+                BingTile.fromCoordinates(1, 0, 1),
+                BingTile.fromCoordinates(1, 1, 1)
+        ).forEach(addIntersecting);
+
+        List<BingTile> outputTiles = new ArrayList<>();
```

Comment:
nit: What do you think about `ImmutableListBuilder` + counter?

Reply:
I'd be all for it if the builder kept the counter.  As it is, it involves a separate variable that has to be kept in sync, which can lead to errors.

Is the advantage that callers of this function won't be able to mutate the List?